repositories/soldier.repository.ts

```diff
@@ -0,0 +1,24 @@
+import { z } from "zod";
+import { Soldier as MongoSoldier} from "./schemas/soldier.schema";
+import { Soldier, SoldierQuery } from "../services/schemas/soldier.zschema";
+
+type Soldier = z.infer<typeof Soldier>;
+type SoldierQuery = z.infer<typeof SoldierQuery>;
+
+async function insertSoldier(soldier: Soldier) {
+    const result = await MongoSoldier.insertMany(soldier);
+    return result;
+}
+
+async function readSoldierByID(id: string) {
```

Comment:
I would prefer calling **get** or **find** instead of **read**, since its closer to what you are doing with mongo...